packages/grafana-data/src/events/types.ts

```diff
@@ -16,6 +16,7 @@ export interface BusEvent {
 export abstract class BusEventBase implements BusEvent {
   readonly type: string;
   readonly payload?: any;
+  readonly source?: any; // panel
```

Comment:
can we automatically inject this based on which panel it is broadcast from?

Reply:
We could create a wrapper/adapter style event bus where all it did was to add this every payload. 

So let's create a EventBusWithSource that takes in an eventbus and a & source (and all it does is to add the source to the payload, otherwise just a wrapper)